src/instruments/src/PFD/FMA.tsx

```diff
@@ -1,554 +1,1021 @@
+                this.textSub.set('');
+            }
+        } else {
+            this.textSub.set('');
+        }
+    }
+
+    onAfterRender(node: VNode): void {
+        super.onAfterRender(node);
+
+        const sub = this.props.bus.getSubscriber<PFDSimvars>();
+
+        sub.on('speedPreselVal').whenChanged().handle((m) => {
+            this.speedPreselVal = m;
+            this.getText();
+        });
+
+        sub.on('machPreselVal').whenChanged().handle((m) => {
+            this.machPreselVal = m;
+            this.getText();
+        });
+
+        sub.on('athrModeMessage').whenChanged().handle((m) => {
+            this.athrModeMessage = m;
+            this.getText();
+        });
+    }
+
+    render(): VNode {
         return (
-            <text className="FontMedium MiddleAlign Cyan" x="35.434673" y="21.656223">{`SPEED SEL ${text}`}</text>
+            <text class="FontMedium MiddleAlign Cyan" x="35.434673" y="21.656223">{this.textSub}</text>
         );
     }
-    return null;
-};
+}
 
-const B1Cell = () => {
-    const activeVerticalMode = getSimVar('L:A32NX_FMA_VERTICAL_MODE', 'enum');
-
-    let text: string | JSX.Element;
-    let inProtection = false;
-
-    switch (activeVerticalMode) {
-    case VerticalMode.GS_TRACK:
-        text = 'G/S';
-        break;
-    // case 2:
-    //     text = 'F-G/S';
-    //     break;
-    case VerticalMode.GS_CPT:
-        text = 'G/S*';
-        break;
-    // case 4:
-    //     text = 'F-G/S*';
-    //     break;
-    case VerticalMode.SRS:
-    case VerticalMode.SRS_GA:
-        text = 'SRS';
-        break;
-    case VerticalMode.TCAS:
-        text = 'TCAS';
-        break;
-    // case 9:
-    //     text = 'FINAL';
-    //     break;
-    case VerticalMode.DES:
-        text = 'DES';
-        break;
-    case VerticalMode.OP_DES:
-        if (getSimVar('L:A32NX_FMA_EXPEDITE_MODE', 'bool')) {
-            text = 'EXP DES';
-        } else {
-            text = 'OP DES';
+class B1Cell extends ShowForSecondsComponent<CellProps> {
+    private boxClassSub = Subject.create('');
+
+    private boxPathStringSub = Subject.create('');
+
+    private activeVerticalModeSub = Subject.create(0);
+
+    private inProtectionClassSub = Subject.create('Cyan');
+
+    private speedProtectionPathRef = FSComponent.createRef<SVGPathElement>();
+
+    private inModeReversionPathRef = FSComponent.createRef<SVGPathElement>();
+
+    private fmaTextRef = FSComponent.createRef<SVGTextElement>();
+
+    private selectedVS = 0;
+
+    private inSpeedProtection = 0;
+
+    private expediteMode = false;
+
+    private FPA = 0;
+
+    constructor(props: CellProps) {
+        super(props, 10);
+    }
+
+    private getText(): boolean {
+        let text: string;
+        let additionalText: string = '';
+
+        this.isShown = true;
+        switch (this.activeVerticalModeSub.get()) {
+        case 31:
+            text = 'G/S';
+            break;
+        // case 2:
```

Comment:
Same for this and others - while we might not normally support these yet, shouldn't we just enable them so they start working? I don't quite understand why they're commented.

Reply:
This was copied straight from the existing FMA code, but yes we can activate it. Shouldn't cause any harm.

Reply:
Ah, in that case I don't mind then, we can enable it whenever we need it - but we might as well do now :+1: